bson/primitive_codecs.go

```diff
@@ -45,15 +45,29 @@ func (pc PrimitiveCodecs) RegisterPrimitiveCodecs(rb *bsoncodec.RegistryBuilder)
 
 // RawValueEncodeValue is the ValueEncoderFunc for RawValue.
 //
-// Deprecated: Use bson.NewRegistry to get a registry with all primitive encoders and decoders
-// registered.
+// If the RawValue's Type is "invalid" and the RawValue's Value is not empty or
+// nil, then this method will return an error.
+//
+// Deprecated: Use bson.NewRegistry to get a registry with all primitive
+// encoders and decoders registered.
 func (PrimitiveCodecs) RawValueEncodeValue(_ bsoncodec.EncodeContext, vw bsonrw.ValueWriter, val reflect.Value) error {
 	if !val.IsValid() || val.Type() != tRawValue {
-		return bsoncodec.ValueEncoderError{Name: "RawValueEncodeValue", Types: []reflect.Type{tRawValue}, Received: val}
+		return bsoncodec.ValueEncoderError{
+			Name:     "RawValueEncodeValue",
+			Types:    []reflect.Type{tRawValue},
+			Received: val,
+		}
 	}
 
 	rawvalue := val.Interface().(RawValue)
 
+	if rawvalue.Type.String() == "invalid" {
```

Comment:
This is a magic string that depends on the logic in the bsontype.Type's String() method, [here](https://github.com/mongodb/mongo-go-driver/blob/master/bson/bsontype/bsontype.go#L57). Do we want to create a function to contain this logic in the bsontype package? Something like "Type.Valid()"?

Reply:
That seems reasonable to me

Reply:
Adding a "check validity" method sounds like a good idea. I recommend naming it `IsValid` to stick with the convention provided by `reflect` and (some) other methods in the Go driver.